[CI] EqlSearchIT class failing

**Build Scans:**
- [elasticsearch-intake #44377 / 8.19.20_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ekywijkljpjra)
- [elasticsearch-pull-request #166296 / 9.4.4_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/echykbsu2ykeo)
- [elasticsearch-pull-request #166281 / 9.5.0_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/unou6ta57qdz6)
- [elasticsearch-pull-request #166296 / 9.4.4_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/gstoqqpyy5s36)
- [elasticsearch-pull-request #166281 / 9.5.0_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/5u43qreczrtda)
- [elasticsearch-pull-request #166250 / 9.4.4_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/aw4sx7ucsdsmq)
- [elasticsearch-pull-request #166250 / 9.4.4_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/6bx3s6caxdyw4)
- [elasticsearch-pull-request #166219 / 9.4.4_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4d4qtppnv5mn6)
- [elasticsearch-pull-request #166219 / 9.4.4_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ks2lv4gmaiaeq)
- [elasticsearch-pull-request #166210 / 9.5.0_part_5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/q7l6upjtgg5ps)

**Reproduction Line:**
```
undefined
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.eql.qa.mixed_node.EqlSearchIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.xpack.eql.qa.mixed_node.EqlSearchIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
undefined
```

**Issue Reasons:**
- [main] 1 consecutive failures in class org.elasticsearch.xpack.eql.qa.mixed_node.EqlSearchIT
- [main] 1 consecutive failures in step 8.19.20_bwc-snapshots
- [main] 19 consecutive failures in step 9.4.4_part_5_bwc-snapshots
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 29 failures in class org.elasticsearch.xpack.eql.qa.mixed_node.EqlSearchIT (2.9% fail rate in 1000 executions)
- [main] 9 failures in step 9.5.0_part_5_bwc-snapshots (5.6% fail rate in 160 executions)
- [main] 19 failures in step 9.4.4_part_5_bwc-snapshots (100.0% fail rate in 19 executions)
- [main] 14 failures in pipeline elasticsearch-pull-request (8.3% fail rate in 168 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-core-infra.

:eyes:

Intake build failure that caused creating this issue failed for a different reason that what the agent reported.
In the intake build failed because the master node wasn't ready and master node logged this:
```
java.lang.AssertionError: expected [index.time_series.temporality_field] to be private but it was not
```
I believe this issue was fixed in https://github.com/elastic/elasticsearch/pull/154931 
The test wasn't muted and I haven't seen any reoccurrences in the later intake builds.

If that would be the only problem then this issue is not a blocker any more.

I'm going to look into the failure reported by the agent now.

--- a/x-pack/plugin/eql/qa/mixed-node/src/javaRestTest/java/org/elasticsearch/xpack/eql/qa/mixed_node/Clusters.java
+++ b/x-pack/plugin/eql/qa/mixed-node/src/javaRestTest/java/org/elasticsearch/xpack/eql/qa/mixed_node/Clusters.java
@@ -51,6 +51,10 @@ public class Clusters {
             .setting("xpack.watcher.enabled", "false")
             .setting("xpack.ml.enabled", "false")
             .setting("xpack.license.self_generated.type", "trial")
+            // DLM frozen tier serialization is gated on both a feature flag and a transport version, so nodes in a mixed cluster can
+            // disagree on the wire format when their build types differ (snapshot vs release). Disable the flag on every node so
+            // serialization is consistent regardless of build type. See https://github.com/elastic/elasticsearch/issues/153679.
+            .systemProperty("es.dlm_searchable_snapshots_feature_flag_enabled", "false")
             .build();
     }
 }

